core/src/main/scala/kanaloa/dispatcher/Dispatcher.scala

```diff
@@ -172,27 +179,25 @@ case class PushingDispatcher(
 }
 
 object PushingDispatcher {
-
   def props[T: BackendAdaptor](
     name:       String,
     backend:    T,
     rootConfig: Config = ConfigFactory.load()
-  )(resultChecker: ResultChecker)(implicit system: ActorSystem) = {
-    val (settings, reporter) = Dispatcher.readConfig(name, rootConfig)
-    val metricsCollector = MetricsCollector(reporter, settings.performanceSamplerSettings)
+  )(resultChecker: ResultChecker)(implicit statsDClient: Option[StatsDClient]) = {
```

Comment:
I don't want to burden users with manually passing a dependency especially when there are already a number of arguments needed for that method, and this is the main usage of Kanaloa, I want to make it as simple as possible. That being said, I am not happy with using `implicit` here either. The real plan is this ticket https://github.com/iheartradio/kanaloa/issues/178.  So this is only temporarily public API for creating Dispatchers.